File: 5/isbn-10 validation.rb
# https://www.codewars.com/kata/51fc12de24a9d8cb0e000001

def validISBN10 isbn
	isbn[0..8].each_char {|x| return false if ("A".."z").include? x}
	n, s = isbn.split("").each {|o| o.sub! "X", "10"}, 0
	n.size.times {|x| s += (x+1)*(n[x]).to_i}

	s%11 == 0 && (!(n.size != 10) || ((n.count(10) >= 1) && (n[9] != 10)))
end

File: 5/isbn-10 validation.go
// https://www.codewars.com/kata/51fc12de24a9d8cb0e000001

package kata

import (
  "strings"
  "strconv"
)

func ValidISBN10(isbn string) bool {
    s, t := strings.Split(isbn, ""), 0
    if len(isbn) != 10 { return false }
  
    for i, v := range s {
        toI, err := strconv.Atoi(v)

        if err != nil && (v == "X" || v == "x") && i == 9 {
            t += (i+1)*10
        } else if err == nil {
            t += (i+1)*toI
        } else {
            return false
        }
    }

    return t % 11 == 0
}